### DISCUSSION TITLE
Upgrading Aurora Postgresql engine version

### DISCUSSION THREAD

**Discussion by franck102** [aws/aws-cdk#30290] [Q&A]
Hi all,

I am trying to upgrade Postgresql in an existing stack, and I am having trouble figuring out how to set this up in CDK. My existing code looks like this:

```
        this.dbCluster = new rds.DatabaseCluster(this, 'DbCluster', {
            engine: rds.DatabaseClusterEngine.auroraPostgres({
                version: rds.AuroraPostgresEngineVersion.VER_15_2,
            }),
            credentials: rds.Credentials.fromSecret(this.postgresSecret, "postgres"),
            defaultDatabaseName: ***,
            vpc: props.vpc,
            serverlessV2MinCapacity: 0.5,
            serverlessV2MaxCapacity: 2,
            securityGroups: [dbSecurityGroup],
            vpcSubnets: props.vpc.selectSubnets({
                subnetType: ec2.SubnetType.PRIVATE_ISOLATED,
            }),
            writer: rds.ClusterInstance.serverlessV2("writer", {
                autoMinorVersionUpgrade: true,
                publiclyAccessible: false,
                enablePerformanceInsights: true,
            }),
            port: this.dbPort,
        })

```

Postgresql 16.2 is supported in the us-east-1 region, so I tried to change my code to:

```
       // create a db cluster
        const dbVersion : rds.AuroraPostgresEngineVersion =
            rds.AuroraPostgresEngineVersion.of("16.2", "16.2");
        
        // https://github.com/aws/aws-cdk/issues/20197#issuecomment-1117555047
        this.dbCluster = new rds.DatabaseCluster(this, 'DbCluster', {
            engine: rds.DatabaseClusterEngine.auroraPostgres({
                version: dbVersion,
            }),
```

... and when I try to deploy I am getting this error:
`
The stack named XxxDatabase248B40BE failed to deploy: UPDATE_ROLLBACK_COMPLETE: Resource handler returned message: "DBClusterParameterGroup not found: default.aurora-postgresql16.2 (Service: Rds, Status Code: 404, Request ID: f7d9b870-34db-4931-b05c-328a926df03c)" (RequestToken: 871870cc-309c-6330-d07b-50f7550d720a, HandlerErrorCode: NotFound)`

Any pointers?
Thanks!
Franck

**Comment by venkatabinary** [comment]
Hi
Here is one way to address this.
It is an error message related to missing v16 Default Aurora PostgreSQL DBClusterParameterGroup.
Create a dummy db instance on v16 with default values. It will implicitly generate the v16 Default Aurora PostgreSQL DBClusterParameterGroup. Now Drop the dummy db instance.
Now try your original code. It should pickup the default DBClusterParameterGroup
Please inform how it goes.
